api/v1alpha1/webhooks/release/webhook.go

```diff
@@ -34,6 +37,29 @@ type Webhook struct {
 	log    logr.Logger
 }
 
+func (w *Webhook) Default(ctx context.Context, obj runtime.Object) error {
+	release := obj.(*v1alpha1.Release)
+
+	if release.Spec.GracePeriodDays == 0 {
+		releasePlan := &v1alpha1.ReleasePlan{}
+
+		err := w.client.Get(ctx, types.NamespacedName{
+			Name:      release.Spec.ReleasePlan,
+			Namespace: release.Namespace,
+		}, releasePlan)
+
+		if err != nil && errors.IsNotFound(err) {
```

Comment:
although this situation would not happen, I think it is better to avoid some eventual panic. Not sure if this is the best approach. Thoughts?

Reply:
I think this situation could actually happen (just not in the automated case). We already have people creating ReleasePlans for nonexistent applications :sweat_smile: 
In fact, I'd like for you to add a test for this situation too

Reply:
I added one extra test just to make sure the default value for GracePeriodDays actually is 0 when the releasePlan does not exist. Turns out I was redundant by calling "Default" as it is done implicitly every time a release is created, so I just need to check it the results are the ones expected. Please see if you think these tests are enough.

Reply:
I think it is sufficient, yep. Thanks!